### DISCUSSION TITLE
Multi-gpu support?

### DISCUSSION THREAD

**Discussion by jhurlbut** [okio-ai/nendo-platform#53] [Ideas]
Is there any support for multi-gpu systems? It would be cool to be able to distribute the action queue across all nvidia gpus in the system. Also, if there are multiple gpus in the system is there a setting for selecting which one to use during the setup/build process?

**Comment by faradox** [comment]
The setup/build process does not use GPU. It's just building the images _with GPU support baked into them_

As for multi-gpu support, yes we already have that built in. A small fix is contained in https://github.com/okio-ai/nendo-server/pull/9 which changes the behavior of GPU actions to request only a single GPU and not all of them as it did before, such that GPU actions can be processed in parallel on multi-GPU hosts. I will merge that branch later today and then all you have to do to enable parallel GPU actions is to set the environment variable `NUM_GPU_WORKERS` on the host equal to the number of available GPUs in your system before calling `make run(-dev)`. That should give you parallel GPU action processing. As usual, let us know if you encounter any issues.
